themes/src/main/resources/theme/base/admin/resources/js/controllers/realm.js

```diff
@@ -2438,14 +2438,16 @@ module.controller('RealmSMTPSettingsCtrl', function($scope, Current, Realm, real
     }
 });
 
-module.controller('RealmEventsConfigCtrl', function($scope, eventsConfig, RealmEventsConfig, RealmEvents, RealmAdminEvents, realm, serverInfo, $location, Notifications, TimeUnit, Dialog) {
+module.controller('RealmEventsConfigCtrl', function($scope, eventsConfig, RealmEventsConfig, RealmEvents, RealmAdminEvents, realm, serverInfo, $location, Notifications, TimeUnit, Dialog, Realm) {
     $scope.realm = realm;
 
     $scope.eventsConfig = eventsConfig;
 
     $scope.eventsConfig.expirationUnit = TimeUnit.autoUnit(eventsConfig.eventsExpiration);
     $scope.eventsConfig.eventsExpiration = TimeUnit.toUnit(eventsConfig.eventsExpiration, $scope.eventsConfig.expirationUnit);
-    
+    $scope.realm.attributes.adminEventsExpirationUnit = TimeUnit.autoUnit(realm.attributes.adminEventsExpiration);
+    $scope.realm.attributes.adminEventsExpiration = TimeUnit.toUnit(realm.attributes.adminEventsExpiration, $scope.realm.attributes.adminEventsExpirationUnit);
+
```

Comment:
Please note these changes should be also propagated to the new admin console, see https://github.com/keycloak/keycloak-ui.
Cc: @edewit @ssilvert Could you please advise here how to incorporate the changes here?

Reply:
> Please note these changes should be also propagated to the new admin console, see https://github.com/keycloak/keycloak-ui. Cc: @edewit @ssilvert Could you please advise here how to incorporate the changes here?

Added issue in keycloak-ui repo.  We will take care of it right after this is merged.  You can ping me in the issue to let me know when that is done.

https://github.com/keycloak/keycloak-ui/issues/3265